streams/src/main/java/org/apache/kafka/streams/state/internals/CacheFlushListener.java

```diff
@@ -31,4 +34,9 @@
      * @param timestamp   timestamp of new value
      */
     void apply(final K key, final V newValue, final V oldValue, final long timestamp);
+
+    /**
+     * Called when records are flushed from the {@link ThreadCache}
+     */
+    void apply(final Record<K, Change<V>> record);
```

Comment:
Do we want to to deprecate the old apply method and use the new one? If so, could we rename one of them to `applyOld` or `applyNew` to differentiate? 

Additionally, we want a parameter signature for `record`

Reply:
I don't want to deprecate it right now, but as with the rest of the "compatibility mode" changes, the old member should become unused by the time @jeqo is done and we can remove it at that time.